### PR TITLE
Search Fixes + Tests

### PR DESCRIPTION
**Pull request by alvinwan**
Tasks:
- [x] add tests for basic search
- [x] add tests for each flag
- [x] sort all

Issues:
- [x] #485 
- [x] #486  

@soumyabasu @Sumukh Ready.

### CODE REVIEW

**Comment by alvinwan** [comment]
fixes #485 #486

File: server/app/models.py [hunk 1]
```diff
@@ -1178,15 +1178,17 @@ class FinalSubmission(Base):
...
     revision = ndb.KeyProperty(Submission)
     queue = ndb.KeyProperty(Queue)
-    submitter = ndb.KeyProperty(User) # TODO Change to ComputedProperty
+    server_time = ndb.ComputedProperty(lambda q: q.submission.get().server_time)
+    # submitter = ndb.ComputedProperty(lambda q: q.submission.get().submitter.get())
+    submitter = ndb.KeyProperty(User)
     published = ndb.BooleanProperty(default=False)
 
     @property
-    def server_time(self):
```
Comment (Sumukh) [inline server/app/models.py]: I don't think we use FinalSubmission.server_time - but I'm not 100% sure. Is there a reason to remove it?

File: server/tests/integration/test_api.py [hunk 2]
```diff
@@ -517,6 +517,152 @@ def test_create_two_entities(self):
...
+    def test_flag_onlybackup(self):
+        """ Testing if onlybackup flag operates without error """
+        query = '-assignment "%s" -onlybackup %s'
+        self.API.querify(query % (self.assignment_name, 'true'))
+        self.API.querify(query % (self.assignment_name, 'false'))
+        
+
+
+if __name__ == '__main__':
+    unittest.main()
```
Comment (Sumukh) [inline server/tests/integration/test_api.py]: File Endings (Should end in a new line)

File: server/tests/integration/test_api.py [hunk 3]
```diff
@@ -517,6 +517,152 @@ def test_create_two_entities(self):
...
+        """ Translate to eq if no operator specified """
+        query = '-assignment Scheme'
+        scope = self.API.translate(query)
+        operator, arg = scope['assignment']
+        self.assertEqual(operator, op.__eq__)
+        self.assertEqual(arg, 'Scheme')
+
+    ###########
+    # QUERIFY #
+    ###########
```
Comment (Sumukh) [inline server/tests/integration/test_api.py]: This is a good start on testing the querying process. If we want to take it further, we would want to verify the results.

**Comment by alvinwan** [comment]
@Sumukh Added more tests that actually verify `-onlyfinal` and `-onlybackup` functionality. Ready.

**Pull request merged**